pkg/plugin/plugin.go

```diff
@@ -173,6 +173,12 @@ var validPluginName = regexp.MustCompile("^[A-Za-z0-9_-]+$")
 
 // validatePluginData validates a plugin's YAML data.
 func validatePluginData(plug *Plugin, filepath string) error {
+	if plug == nil {
+		return fmt.Errorf("nil plugin value detected")
```

Comment:
Since users will see this error message, we probably want something more informative than “nil”. It’s not very informative to the user - how are they supposed to rectify this?

What real-world situation can arise where plug will return nil?